src/sentry/issues/occurrence_consumer.py

```diff
@@ -127,17 +156,30 @@ def _get_kwargs(payload: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
         return None
 
 
+def _validate_kwargs(kwargs: Mapping[str, Any]) -> None:
+    # only validate event data, for now
+    try:
+        jsonschema.validate(kwargs["event_data"], EVENT_PAYLOAD_SCHEMA)
+    except jsonschema.exceptions.ValidationError:
+        metrics.incr("occurrence_ingest.event_payload_invalid")
+        raise InvalidEventPayloadError("Event payload does not match schema")
+
+
 def _process_message(
     message: Mapping[str, Any]
 ) -> Optional[Tuple[IssueOccurrence, Optional[GroupInfo]]]:
-    kwargs = _get_kwargs(message)
+    metrics.incr("occurrence_ingest.messages", sample_rate=1.0)
```

Comment:
Do you need full sample rate if you're in an ingest adjacent consumer?

Reply:
For now we do, since this will be very low volume